lib/providers/quoters/RoutingApiQuoter.ts

```diff
@@ -22,12 +30,76 @@ export class RoutingApiQuoter implements Quoter {
+        // and we simply don't account for portion and let quote request processing continue
+        log.error({ e }, 'Failed to resolve tokenIn & tokenOut');
+        metrics.putMetric(`PortionProvider.resolveTokenErr`, 1);
+      }
+
+      const portion = (
+        await this.portionProvider.getPortion(
+          request.info,
+          resolvedTokenIn?.wrapped.address,
+          resolveTokenOut?.wrapped.address
+        )
+      ).portion;
+
+      const req = this.buildRequest(request, portion, resolveTokenOut);
       const now = Date.now();
-      const response = await axios.get(req, { headers: { 'x-api-key': this.routingApiKey } });
+      const response = await axios.get<ClassicQuoteDataJSON>(req, { headers: { 'x-api-key': this.routingApiKey } });
+
+      // we need to use raw quote amount for exact in, not quote adjusted gas amount, because raw quote amount is more accurate to estimate
+      // the portion amount
+      // although clients are expected to use the portionBips for exact in swaps for the best accurate, i.e. tokenOutAmount not important for exact in
+      // tokenOutAmount only important for exact out, which is requested amount
+      const tokenOutAmount =
+        request.info.type === TradeType.EXACT_OUTPUT ? request.info.amount.toString() : response.data.quote;
+      const portionAmount = this.portionProvider.getPortionAmount(tokenOutAmount, portion, resolveTokenOut);
+      const portionAdjustedQuote = this.portionProvider.getPortionAdjustedQuote(
+        request.info,
+        response.data.quote,
+        response.data.quoteGasAdjusted,
+        portionAmount,
+        resolvedTokenIn,
+        resolveTokenOut
+      );
+      const quoteGasAndPortionAdjusted = portionAdjustedQuote?.quotient.toString() ?? response.data.quoteGasAdjusted;
+      const quoteGasAndPortionAdjustedDecimals =
+        portionAdjustedQuote?.toExact() ?? response.data.quoteGasAdjustedDecimals;
+
+      const portionAdjustedResponse: AxiosResponse<ClassicQuoteDataJSON> = {
+        ...response,
+        data: ENABLE_PORTION(process.env.ENABLE_PORTION)
+          ? {
+              ...response.data,
+              portionBips: portion?.bips, // important for exact in, clients are expected to use this for exact in swaps
+              portionRecipient: portion?.recipient, // important, clients are expected to use this for exact in and exact out swaps
+              portionAmount: portionAmount?.quotient.toString(), // important for exact out, clients are expected to use this for exact out swaps
+              portionAmountDecimals: portionAmount?.toExact(), // important for exact out, clients are expected to use this for exact out swaps
+              quoteGasAndPortionAdjusted: quoteGasAndPortionAdjusted, // not important, clients disregard this
+              quoteGasAndPortionAdjustedDecimals: quoteGasAndPortionAdjustedDecimals, // not important, clients disregard this
```

Comment:
I'm not too sure about marking these fields as unimportant. Just feel like that pattern could be misleading. For example, the UniswapX bot we wrote relied on a bunch of fields the interface maybe didn't use but we still expected them to be accurate/reliable. Maybe this can get turned into an offline discussion because I just might be missing some context here. Don't want to block this PR any longer as I know we have some time constraints but would be nice to have a follow up here.